backend/metadata_writer/src/metadata_writer.py

```diff
@@ -325,8 +333,14 @@ def is_kfp_v2_pod(pod) -> bool:
 
                     output_artifacts = []
                     for name, art in argo_output_artifacts.items():
-                        artifact_uri = argo_artifact_to_uri(art)
-                        if not artifact_uri:
+                        artifact_uri_check = argo_artifact_to_uri(art)
+                        if artifact_uri_check:
+                            if re.search('(\W+)', artifact_uri_check).group(1) != '://':
+                                artifact_uri_wo_key = argo_artifact_to_uri(argo_template.get('archiveLocation', {}), True)
```

Comment:
Maybe we can always pass `argo_template.get('archiveLocation', {})` to `argo_artifact_to_uri` and encapsulate this logic there?

Reply:
The thing is, that despite the property name in `archiveLocation` they're storing the folder of artifact location but not the actual archive (file) location. For example in the `outputs` you will have file location like `"key":"artifacts/artifact-passing-lr6fg/artifact-passing-lr6fg-2908156709/hello-art.tgz"` and within `archiveLocation` you will have something like `"key":"artifacts/artifact-passing-lr6fg/artifact-passing-lr6fg-2908156709"`